File: flyrxbus/src/main/java/com/xia/flyrxbus/TagMessage.java
package com.xia.flyrxbus;

@SuppressWarnings("WeakerAccess")
final class TagMessage {
    public Object mEvent;
    public String mTag;

    public TagMessage(Object event, String tag) {
        mEvent = event;
        mTag = tag;
    }

    public boolean isSameType(final Class eventType, final String tag) {
        return Utils.equals(getEventType(), eventType)
                && Utils.equals(this.mTag, tag);
    }

    public Class getEventType() {
        return Utils.getClassFromObject(mEvent);
    }

    @Override
    public String toString() {
        return "event: " + mEvent + ", tag: " + mTag;
    }
}

File: flyrxbus/src/main/java/com/xia/flyrxbus/TagMessage.kt
package com.xia.flyrxbus

import androidx.annotation.NonNull
import io.reactivex.internal.functions.ObjectHelper

class TagMessage(@NonNull @JvmField var mEvent: Any, @NonNull @JvmField var mTag: String) {

    fun isSameType(eventType: Class<*>, tag: String): Boolean {
        return ObjectHelper.equals(getEventType(), eventType) && ObjectHelper.equals(this.mTag, tag)
    }

    fun getEventType(): Class<*>? {
        return Utils.getClassFromObject(mEvent)
    }

    override fun toString(): String {
        return "event: $mEvent, tag: $mTag"
    }
}
